sql-plugin/src/main/scala/com/nvidia/spark/rapids/RegexParser.scala

```diff
@@ -2007,4 +2007,99 @@ class RegexUnsupportedException(message: String, index: Option[Int])
       case _ => message
     }
   }
-}
\ No newline at end of file
+}
+
+sealed trait RegexOptimizationType
+object RegexOptimizationType {
+  case class Equals(pattern: String) extends RegexOptimizationType
+  case class StartsWith(pattern: String) extends RegexOptimizationType
+  case class EndsWith(pattern: String) extends RegexOptimizationType
+  case class Contains(pattern: String) extends RegexOptimizationType
+  case object NoOptimization extends RegexOptimizationType
+}
+
+object RegexRewriteUtils {
+  private def isSimplePattern(astLs: List[RegexAST]): Boolean = {
+    astLs.forall {
+      case RegexChar('^') | RegexChar('$') => false
+      case RegexChar(_) => true
+      case _ => false
+    }
+  }
+
+  private def isWildcard(ast: RegexAST): Boolean = {
+    ast match {
+      case RegexRepetition(RegexChar('.'), SimpleQuantifier('*')) => true
+      case RegexSequence(parts) if parts.forall(isWildcard) => true
+      case RegexGroup(_, term, _) if isWildcard(term) => true
+      case _ => false
+    }
+  }
+
+  private def stripLeadingWildcards(astLs: List[RegexAST]): List[RegexAST] = astLs match {
+    case RegexChar('^') :: _  =>
+      astLs.drop(1).dropWhile(isWildcard)
+    case _ => astLs.dropWhile(isWildcard)
+  }
+
+  private def stripTailingWildcards(astLs: List[RegexAST]): List[RegexAST] = {
+    if (astLs.lastOption == Some(RegexChar('$'))) {
+      astLs.dropRight(1).reverse.dropWhile(isWildcard).reverse
+    } else {
+      astLs.reverse.dropWhile(isWildcard).reverse
+    }
+  }
+
+  private def RegexCharsToString(chars: List[RegexAST]): String = {
+    chars.map {
+      case RegexChar(ch) => ch
+      case _ => throw new IllegalArgumentException("Invalid character")
+    }.mkString
+  }
+
+  def matchSimplePattern(ast: RegexAST): RegexOptimizationType = {
+    ast.children().toList match {
+      case RegexChar('^') :: RegexGroup(_, RegexSequence(parts), None) :: List(RegexChar('$')) 
+          if isSimplePattern(parts.toList) => {
+        // ^(pattern)$ => Equals pattern
+        RegexOptimizationType.Equals(RegexCharsToString(parts.toList))
+      }
+      case RegexChar('^') :: rest if rest.last == RegexChar('$') && isSimplePattern(rest.init) => {
+        // ^pattern$ => Equals pattern
+        RegexOptimizationType.Equals(RegexCharsToString(rest.init))
```

Comment:
same repeated init comment as before
```suggestion
      case (RegexChar('^') :: restInit) :+ RegexChar('$') if isSimplePattern(restInit) => {
        // ^pattern$ => Equals pattern
        RegexOptimizationType.Equals(RegexCharsToString(restInit))
```

Reply:
Removed this case now for another issue.